binding/button.go

```diff
@@ -0,0 +1,39 @@
+package binding
+
+import (
+	"fyne.io/fyne"
+	"fyne.io/fyne/widget"
+)
+
+func BindButtonTapped(button *widget.Button, data *BoolBinding) {
```

Comment:
Wrapping on-tapped like this is dangerous because it only works if it's called after the custom OnTapped function is set.
I'm not sure that a data binding to a user action anyway - developers can just call data.Set(true) in their OnTapped ;)

Reply:
Would you agree that the reason to use 2-way data binding is so that if the data changes the UI is updated _and_ if the UI is changed the data is updated?

Reply:
For elements that can change data, then yes 2 way binding's aim is to update data appropriately.
When discussing this with @steveoc64 I think we came to the conclusion that we can't use public hooks because it causes confusion. If a developer changes OnTapped or OnChanged (which they are entitled to do) this breaks.

Just as I suggested widgets could have a simple .Bind(type) method that calls out to data binding code I don't think it would be a big deal for a widget to call out to databinding "ValueChanged(...)" or similar before it calls the user hooks. That way it does not impact the previous function of callbacks.